vulkaninfo/vulkaninfo.h

```diff
@@ -549,24 +553,38 @@ struct SurfaceExtension {
     }
 };
 
-struct VulkanVersion {
-    uint32_t major = 1;
-    uint32_t minor = 0;
-    uint32_t patch = 0;
-    VulkanVersion() = default;
-    VulkanVersion(uint32_t version) noexcept
-        : major{VK_VERSION_MAJOR(version)}, minor{VK_VERSION_MINOR(version)}, patch{VK_VERSION_PATCH(version)} {}
-    std::string str() { return std::to_string(major) + "." + std::to_string(minor) + "." + std::to_string(patch); }
+class APIVersion {
+  public:
+    APIVersion() : api_version_(VK_API_VERSION_1_0) {}
+    APIVersion(uint32_t api_version) : api_version_(api_version) {}
+    APIVersion& operator=(uint32_t api_version) {
+        api_version_ = api_version;
+        return *this;
+    }
```

Comment:
I find the inclusion of the copy operator strange, since it shouldn't be required here. 
This type can use rule of zero since it has no special copy/move semantics.

Reply:
Fair point, I don't think this is needed for Vulkan anyway, I just had it there for symmetry, because on the Vulkan SC side it has custom code, but it makes sense to not include it in the Vulkan code. I've removed it.